### PR TITLE
Number

### PR DESCRIPTION
**Pull request by kofi-clever**
**Jira:** https://clever.atlassian.net/browse/FEE-58

**Overview:**
Adding a `Number` component for consistently formatting (and optionally shortening) long numbers for display.

**Screenshots/GIFs:**
number

**Testing:**
- [x] Unit tests
- Manual tests:
  - [x] Chrome
  - [x] Safari
  - [x] IE10

**Roll Out:**
- Before merging:
  - [x] Updated docs
  - [x] Bumped version in `package.json`
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -19,3 +21,4 @@ export {Tooltip} from "./Tooltip";
 export {Label} from "./Label";
 export {FileInput} from "./FileInput/FileInput";
 export * from "./InfoPanel";
+export {Number};
```
Comment (ahsojar) [inline src/index.js]: haven't written a component before, why can't we do `export {Number} from "./Number"` like we do for the lines above?

File: test/Number_test.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  it("supports non-React formatting", () => {
+    assert.equal("5,123", Number.format(5123));
+    assert.equal("6.8M", Number.format("6789345", true));
+  });
+
+  it("throws for non-numeric input", () => {
+    assert.throws(() => shallow(<Number>nope</Number>));
+    assert.throws(() => Number.format("nope", true));
+  });
+});
```
Comment (ahsojar) [inline test/Number_test.jsx]: Nice tests 🎉

File: src/index.js (hunk 1 above)
Comment (kofi-clever) [inline src/index.js]: it's a quirk of the es6 import/export pattern. `Number` is currently a default export, so it has to be imported as a default here and then exported as a secondary. (the others are exported as secondary imports, but i'm toying with the idea of providing default exports so clients can import individual components without pulling all of the components in)

Comment (ahsojar) [inline src/index.js]: ahh gotcha. perhaps worth adding a comment for `Number` since its a special case!

File: test/Number_test.jsx (hunk 2 above)
Comment (kofi-clever) [inline test/Number_test.jsx]: +1

**Pull request merged**